### ISSUE TITLE
The RadioControl component renders invalid and not accessible markup

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52890]
### Description

While investigating the Post Status UI in teh Site editor in https://github.com/WordPress/gutenberg/pull/52885 and https://github.com/WordPress/gutenberg/pull/52634 I noticed the `RadioControl` component is open to misuse by developers and has some limitations that cause accessibility problems. Both the component and its implementations should be reviewed thoroughly.

At a first check:

- The `label` prop renders a stray label element that is not associated with any input.
- No fieldset and legend to group the radio boxes.
- No ability to set a description for each radio button (associated with aria-describedby).
- As such, in the `PageStatus` component the description has been added within the label, thus making the accessible name of the inputs very confusing: name and description should be separated.
- The `help` prop renders some additional text at the end of all radio buttons. This text is associated with all the radio buttons via aria-describedby. This may be useful in some cases but it's likely to be redundant and confusing. In most cases, each radio button will need its won description. Worth reminding aria-describedby may accept multiple ID refs.

In the screenshot below I made some visually hidden elements visible to better illustrate, and set some options:

- The RadioControl component [in use in the Site editor `PageStatus`](https://github.com/WordPress/gutenberg/blob/dc27d6b034a0d112dce0b71fb7d3d5dd8e870fe6/packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-status.js#L187-L194).
- The [code example provided in the component inline documentation](https://github.com/WordPress/gutenberg/blob/dc27d6b034a0d112dce0b71fb7d3d5dd8e870fe6/packages/components/src/radio-control/index.tsx#L27-L42), actually rendered.
- The component [in the Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-radiocontrol--default&args=help:This+is+the+help+text;hideLabelFromVision:false).

In all three case the HTML is invalid. The labelling isn't ideal, given the stray label.

Given the component limitations, maybe it's not a case that in the Post editor, in the very similar Post visibility UI, an ad-hoc implementation has been made, see the [PostVisibilityChoice component](https://github.com/WordPress/gutenberg/blob/dc27d6b034a0d112dce0b71fb7d3d5dd8e870fe6/packages/editor/src/components/post-visibility/index.js#L135-L161).

Ideally, the `RadioControl` component should behave more like the PostVisibilityChoice one, or at least produce valid markup, render fieldset+legend, and be more flexible to be able to replace PostVisibilityChoice.

Cc @ciampo @mirka 

### Step-by-step reproduction instructions

- Go to the storybook https://wordpress.github.io/gutenberg/?path=/story/components-radiocontrol--default&args=help:This+is+the+help+text;hideLabelFromVision:false
- Observe the `POST VISIBILITY` text is actually a label element that is not associated with any input.
- Make sure to enter some help text.
- Observe the help text is rendered dat the bottom and it's associated with _all_ radio buttons via aria-decribedby.
- Observe here is no way to set a specific description for a specific radio button.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mirka** [comment]
For quick reference, here are the radio UI patterns used in the Post Visibility and Page Status components:

<img width="269" alt="Post Visibility radio" src="https://github.com/WordPress/gutenberg/assets/555336/ead040bc-6a5e-4a78-9ab2-33ff1eb1e538">

<img width="310" alt="Page Status radio" src="https://github.com/WordPress/gutenberg/assets/555336/d3a2eb5d-6c2e-4830-99a2-0ee9f31e62d9">

RadioControl does not support this pattern yet, and the above usages work around that in different ways. (Post Visibility doesn't even use RadioControl and does a custom implementation.)

I classify this as a similar problem to [what we've seen with CheckboxControl](https://github.com/WordPress/gutenberg/issues/51740#issuecomment-1853473714), where our components aren't flexible enough therefore it's very easy for consumers to overlook certain accessibility considerations in their workarounds.

I think we should look into providing an easier, "officially sanctioned" way to accomplish these kinds of custom layouts with the (currently monolithic) CheckboxControl/RadioControl components so we can provide better documentation and examples on how to do it accessibly. It might look similar to how we're designing the new `Tabs` and `DropdownMenu` components, where we export all the subcomponents to support customization.
